File: OP/OPlaba5/C++/OPlaba5/OPlaba5/OPlaba5.cpp
#include<iostream>
#include<cstdlib>
#include<cmath>
#include<float.h>
using namespace std;

int main()
{
    float a;
    int n = 2;
    int max = 0;
    int min = 0;
    cout << "Enter the number: ";
    cin >> a;
    int i = 2;
    if (a < 2)
    {
        cout << "The nearest max: 2" << endl;
    }
    else
    {
        while (max <= a)
        {
            for (i = 2; n; i += 1)
            {
                if (n % i == 0)
                {
                    max = i;
                    break;
                }
            }
            n += 1;
        }
        n = a;
        while (min != i)
        {
            for (i = 2; n; i += 1)
            {
                if (n % i == 0)
                {
                    min = n;
                    break;
                }
            }
            n -= 1;
        }
        if (max - a > a - min)
        {
            cout << "The nearest min: ""The nearest min: " << min << endl;
        }
        else
        {
            cout << "The nearest max: " << max << endl;
        }
    }
}

File: OP/OPlaba5/OPlaba5/Python/OPlaba5/OPlaba5/OPlaba5.py
a=float(input("Enter the number: "))
n=2
max=0
min=0
i=2
if a<2:
    print("The nearest max: 2")
else:
    while max<=a:
        for i in range (2, n+1, 1):
            if n%i==0:
                max=i
                break
        n+=1
    n=int(a)
    while min!=i:
        for i in range (2, n+1, 1):
            if n%i==0:
                min=n
                break
        n-=1
    if max-a>a-min:
        print("The nearest min: ", min)
    else:
        print("The nearest max: ", max)